### PR TITLE
Rename `*Descriptor` -> `*Type` to match Wasm spec

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Renaming things to be better aligned with the Wasm spec.

We'll be adding compatibility definitions with a deprecated attribute where possible.

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

File: tests/high_level_api.rs [hunk 1]
```diff
@@ -87,36 +87,36 @@ wasmer_backends! {
...
+                    ty: export::ExternType::Table(TableType {
                         element: ElementType::Anyfunc,
                         minimum: 2,
                         maximum: None,
                     }),
                 },
-                export::ExportDescriptor {
+                export::ExportType {
                     name: "test-global",
-                    ty: export::ExternDescriptor::Global(GlobalDescriptor {
```
Comment (nlewycky) [inline tests/high_level_api.rs:100]: This field `ty` should be named `desc` for descriptor. The two places descriptors show up in the spec are as part of export types and import types (see 2.5.10 and 2.5.11). The export type is {name, desc} and the desc is {func | table | mem | global}.

Comment (MarkMcCaskey) [reply tests/high_level_api.rs:100]: Very good catch!  I just discussed the use of `desc` with Syrus now and we looked at the Wasm C API was doing as a reference and it seems that it just generally avoids indices in favor of other types, like `wasm_exporttype` here (https://github.com/WebAssembly/wasm-c-api/blob/master/include/wasm.h#L295).  So we're going to defer this for now, but this is something worth following up on.

**Comment by MarkMcCaskey** [comment]
bors try

File: lib/runtime-core/src/error.rs [hunk 2]
```diff
@@ -100,37 +100,37 @@ pub enum LinkError {
         name: String,
     },
     /// The memory descriptor provided does not match the expected descriptor.
-    IncorrectMemoryDescriptor {
+    IncorrectMemoryType {
```
Comment (MarkMcCaskey) [inline lib/runtime-core/src/error.rs:103]: Something I want reviewers to note about this PR is that it introduces breaking changes in `LinkError`, the error variants are renamed here:  I don't see a good way to make this change without breaking these.  Well, unless we're fine with the errors referring to them as `descriptors`

**Comment by syrusakbary** [comment]
After talking offline, we will delay merging this PR to batch other breaking API changges

**Comment by syrusakbary** [comment]
This issue should be fixed in wasmer `master` branch. Closing PR!

**Pull request closed without merging**